src/GraphQL.Tests/CoreToVanillaConverterTests.cs

```diff
@@ -0,0 +1,94 @@
+using System.Threading.Tasks;
+using GraphQL.Language;
+using GraphQL.Types;
+using GraphQLParser.AST;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Bugs
+{
+    // https://github.com/graphql-dotnet/graphql-dotnet/pulls/1781
+    public class CoreToVanillaConverterTests : QueryTestBase<PR1781Schema>
+    {
+        [Fact]
+        public async Task DocumentExecuter_really_big_double_valid()
+        {
+            var de = new DocumentExecuter();
+            var valid = await de.ExecuteAsync(new ExecutionOptions
+            {
+                Query = $"{{ test(arg:{double.MaxValue:0}0.0) }}",
```

Comment:
```suggestion
                // create a floating-point value that is larger than double.MaxValue
                // in the expression "{double.MaxValue:0}0.0" below, the 0.0 effectively
                //   multiplies double.MaxValue by 10 and the .0 forces the parser to
                //   assume it is a floating point value rather than a large integer
                Query = $"{{ test(arg:{double.MaxValue:0}0.0) }}",
```